opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java

```diff
@@ -426,4 +430,68 @@ public void run() {
             }
         });
     }
+
+    private String[] getStructureColumns() {
+        return new String[]{
+                TASK_TABLE + "." + ID,
+                TASK_TABLE + "." + CODE,
+                TASK_TABLE + "." + FOR,
+                TASK_TABLE + "." + BUSINESS_STATUS,
+                TASK_TABLE + "." + STATUS,
+                TASK_TABLE + "." + STRUCTURE_ID
+        };
+    }
+
+    public String getTaskSelect(String mainCondition) {
+        SmartRegisterQueryBuilder queryBuilder = new SmartRegisterQueryBuilder();
+        queryBuilder.selectInitiateMainTable(TASK_TABLE, getStructureColumns(), ID);
+        return queryBuilder.mainCondition(mainCondition);
+    }
+
+    public void resetInterventionTaskInfo(Context context, String interventionType, String featureId) {
+        String sql = String.format(getTaskSelect("%s = ? and %s = ?"),
+                FOR, CODE);
+
+        final String SQL = sql;
+
+        appExecutors.diskIO().execute(() -> {
+            StructureTaskDetails taskDetails = null;
+            Cursor cursor = null;
+            boolean taskInfoResetSuccessful = false;
+            try {
+                cursor = getDatabase().rawQuery(SQL, new String[]{featureId, interventionType});
+                if (cursor.moveToNext()) {
+                    taskDetails = readTaskDetails(cursor);
+                    // Reset task info
+                    interactorUtils.resetTaskInfo(context, getDatabase(), taskDetails);
```

Comment:
Shouldn't this be executed once you have closed this cursor. This means the cursor is left open and other cursors are opened in `resetTaskInfo` means a high change of database locks